src/main.rs

```diff
@@ -9,11 +9,18 @@ mod visualization;
 
 use client::run_client;
 use game_state::GameState;
+use game_state::GameAction;
+use resource::Resource;
+use enum_iterator::IntoEnumIterator;
 
 fn main() {
     let mut state = GameState::new();
     state.register_player();
     state.register_player();
 
+    for res in Resource::into_enum_iter() {
+        state.handle_action(GameAction::AllocateWorker(1, res));
+    };
```

Comment:
It doesn't matter here but the semicolon doesn't actually do anything. The broader point that might be useful to bring up is that cargo has a built-in formatter, which I tend to run either before each commit or before merge, and it has a few small changes to make in this file including this one :)

Reply:
Another thing that is quite helpful (but that doesn't have anything to say about your great code here) is the somewhat standard linter [`clippy`](https://github.com/rust-lang/rust-clippy)

```
cargo clippy
```

Reply:
Thanks for the tips! Ran the auto-formatter and made another commit